src/network/http/websocket.cpp

```diff
@@ -197,25 +191,25 @@ namespace fc { namespace http {
       {
          public:
             websocket_server_impl()
-            :_server_thread( fc::thread::current() )
+               :_server_thread( fc::thread::current() ), fwd_header_key("")
             {
-
                _server.clear_access_channels( websocketpp::log::alevel::all );
                _server.init_asio(&fc::asio::default_io_service());
                _server.set_reuse_addr(true);
                _server.set_open_handler( [&]( connection_hdl hdl ){
-                    _server_thread.async( [&](){
-                       auto new_con = std::make_shared<websocket_connection_impl<websocket_server_type::connection_ptr>>( _server.get_con_from_hdl(hdl) );
-                       _on_connection( _connections[hdl] = new_con );
-                    }).wait();
+                  _server_thread.async( [&](){
+                        auto new_con = std::make_shared<websocket_connection_impl<websocket_server_type::connection_ptr>>( _server.get_con_from_hdl(hdl) );
+                        _on_connection( _connections[hdl] = new_con );
+                     }).wait();
                });
                _server.set_message_handler( [&]( connection_hdl hdl, websocket_server_type::message_ptr msg ){
                     _server_thread.async( [&](){
                        auto current_con = _connections.find(hdl);
                        assert( current_con != _connections.end() );
-                       wdump(("server")(msg->get_payload()));
                        auto payload = msg->get_payload();
                        std::shared_ptr<websocket_connection> con = current_con->second;
+                       wlog( "Websocket Server Remote: ${host} Payload: ${body}", 
+                           ("host", con->get_remote_hostname(fwd_header_key)) ("body", msg->get_payload()));
```

Comment:
Can we save `get_remote_hostname()` to a member variable to avoid re-extracting on every request?

Reply:
I am caching as part of the connection object. As always, I am open for feedback.